catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/content/brick/app/WhenBrickTest.java

```diff
@@ -0,0 +1,152 @@
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.catrobat.catroid.uiespresso.content.brick.app;
+
+import org.catrobat.catroid.content.Script;
+
+import org.catrobat.catroid.R;
+import org.catrobat.catroid.content.bricks.BroadcastReceiverBrick;
+import org.catrobat.catroid.content.bricks.PlaceAtBrick;
+import org.catrobat.catroid.content.bricks.WhenBackgroundChangesBrick;
+import org.catrobat.catroid.content.bricks.WhenClonedBrick;
+import org.catrobat.catroid.content.bricks.WhenConditionBrick;
+import org.catrobat.catroid.content.bricks.WhenStartedBrick;
+import org.catrobat.catroid.content.bricks.WhenTouchDownBrick;
+import org.catrobat.catroid.physics.content.bricks.CollisionReceiverBrick;
+import org.catrobat.catroid.ui.ScriptActivity;
+import org.catrobat.catroid.uiespresso.content.brick.utils.BrickCoordinatesProvider;
+import org.catrobat.catroid.uiespresso.content.brick.utils.BrickTestUtils;
+import org.catrobat.catroid.uiespresso.testsuites.Cat;
+import org.catrobat.catroid.uiespresso.testsuites.Level;
+import org.catrobat.catroid.uiespresso.util.UiTestUtils;
+import org.catrobat.catroid.uiespresso.util.matchers.BrickCategoryListMatchers;
+import org.catrobat.catroid.uiespresso.util.matchers.BrickPrototypeListMatchers;
+import org.catrobat.catroid.uiespresso.util.rules.BaseActivityInstrumentationRule;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import static org.junit.runners.Parameterized.*;
+
+import static android.support.test.espresso.Espresso.onData;
+import static android.support.test.espresso.Espresso.onView;
+import static android.support.test.espresso.action.ViewActions.click;
+import static android.support.test.espresso.matcher.ViewMatchers.withId;
+
+import static org.catrobat.catroid.uiespresso.content.brick.utils.BrickDataInteractionWrapper.onBrickAtPosition;
+import static org.hamcrest.Matchers.allOf;
+import static org.hamcrest.Matchers.instanceOf;
+import static org.hamcrest.core.Is.is;
+
+import java.util.Arrays;
+import android.support.annotation.StringRes;
+
+
+@RunWith(Parameterized.class)
+public class WhenBrickTest {
+	@Parameters(name = "{0}")
+	public static Iterable<Object[]> data() {
+		return Arrays.asList(new Object[][] {
+			{1,R.string.brick_when_started,WhenStartedBrick.class},
+			{2,R.string.brick_when_becomes_true,WhenConditionBrick.class},
+			{3,R.string.brick_when_cloned,WhenClonedBrick.class},
+			{4,R.string.brick_when_background,WhenBackgroundChangesBrick.class},
+			{5,R.string.brick_when_touched,WhenTouchDownBrick.class},
+			{6,R.string.brick_broadcast_receive,BroadcastReceiverBrick.class},
+			{7,R.string.brick_collision_receive,CollisionReceiverBrick.class}
+		});
+	}
+
+	@Parameter
+	public int testId;
```

Comment:
1,2,3,4,5 as name for the test is rather bad.
please use the classes name instead.